Flaky integration tests in main module

The main module tests relies on `Vault` launched using `testingContainers` for integration tests. These seem to be flaky.

Note that we are using testing containers but with PostgreSQL in the backend and we've not noticed any flakiness, so I'd suggest to make sure the healthchecks and retries are properly set in this case.

```
--- FAIL: TestVaultIntegration (14.17s)
    --- FAIL: TestVaultIntegration/TestDeleteCreds (7.87s)
        keyval_test.go:125: 
            	Error Trace:	/home/runner/work/chainloop/chainloop/internal/credentials/vault/keyval_test.go:125
            	Error:      	Received unexpected error:
            	            	validating client: error writing secret to secret/data//chainloop-healthcheck: context deadline exceeded
            	Test:       	TestVaultIntegration/TestDeleteCreds
FAIL
FAIL	github.com/chainloop-dev/chainloop/internal/credentials/vault	14.177s
ok  	github.com/chainloop-dev/chainloop/internal/ociauth	0.003s
ok  	github.com/chainloop-dev/chainloop/internal/robotaccount/cas	0.035s
```

--- a/internal/credentials/vault/keyval_test.go
+++ b/internal/credentials/vault/keyval_test.go
@@ -203,7 +203,7 @@ func newVaultInstance(t *testing.T) *vaultInstance {
 		Env: map[string]string{
 			"VAULT_DEV_ROOT_TOKEN_ID": defaultToken,
 		},
-		WaitingFor: wait.ForListeningPort("8200/tcp"),
+		WaitingFor: wait.ForHTTP("/v1/sys/health").WithPort("8200/tcp"),
 	}
 
 	instance, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
@@ -211,8 +211,7 @@ func newVaultInstance(t *testing.T) *vaultInstance {
 		Started:          true,
 	})
 
-	assert.NoError(t, err)
-
+	require.NoError(t, err)
 	return &vaultInstance{instance}
 }
 
@@ -242,9 +241,7 @@ func (s *testSuite) SetupTest() {
 }
 
 func (s *testSuite) TearDownTest() {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-	defer cancel()
-	assert.NoError(s.T(), s.vault.instance.Terminate(ctx))
+	assert.NoError(s.T(), s.vault.instance.Terminate(context.Background()))
 }
 
 // Run the tests

